src/cargo/ops/cargo_remove.rs

```diff
@@ -59,7 +91,76 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
             .warn("aborting remove due to dry run")?;
     } else {
         manifest.write()?;
+
+        if let Some(parent_manifest) = parent_manifest {
+            cargo_util::paths::write(
+                options.workspace.root_manifest(),
+                parent_manifest.to_string().as_bytes(),
+            )?;
+        }
     }
 
     Ok(())
 }
+
+/// Get whether or not a dependency is marked as `workspace`.
+fn dependency_is_workspace(dep: &str, dep_table: &[String], manifest: &LocalManifest) -> bool {
+    if let Ok(toml_edit::Item::Table(table)) = manifest.get_table(dep_table) {
+        let value = table.get(dep).and_then(|i| i.get("workspace"));
+        if let Some(toml_edit::Item::Value(value)) = value {
+            return value.as_bool() == Some(true);
+        }
+    }
+
+    false
+}
+
+/// Get whether or not a dependency is depended upon in a workspace.
+fn dependency_in_workspace(
+    dep: &str,
+    dep_table: &DepTable,
+    exclude: &Package,
+    workspace: &Workspace<'_>,
+) -> bool {
+    workspace.members().any(|p| {
+        let manifest = LocalManifest::try_new(p.manifest_path()).unwrap(); // TODO handle error
+        p.dependencies()
+            .iter()
+            .filter(|d| d.package_name().as_str() == dep)
```

Comment:
That's a good point, will fix. Now that you mention it, we've already always been removing by the TOML name, which is a little strange.

Reply:
> Now that you mention it, we've already always been removing by the TOML name, which is a little strange.

`cargo add` also generally works with the dependency name, rather than the package name.  This removes ambiguity when someone has a dependency on two different versions of a package.